runner.py

```diff
@@ -296,6 +296,29 @@ def validate_input(output_dir, input_dir, reference_file, mode):
         raise Exception("Could not find files ending with '.fastq' or 'fastq.gz' in input_dir !")
 
 
+def create_stats_file(output_dir, filenames, overlapping_reads, reads_input_num):
+    file_path = os.path.join(output_dir, "stats_file.stats")
+    called_bases_path = os.path.join(output_dir, "called_bases.tsv")
+    dict_states = {'N': '1', 'M': '1 or 2', 'Y': 'exactly 2'}
+    with open(filenames['read_counter_file'], 'r') as read_counter:
+        counter_pd = pd.read_csv(read_counter, sep="\t", engine='python')
+        len_counter = str(len(counter_pd.read_id))
+        mapped_once = str(len(counter_pd.loc[counter_pd['number_of_alignments'] == 1].read_id))
+        mapped_twice = str(len(counter_pd.loc[counter_pd['number_of_alignments'] == 2].read_id))
+    with open(called_bases_path, 'r') as called_bases_file:
```

Comment:
you should never read called_bases directly as it can be a very large file... 
doing this would mean that the program could not be ran locally...